src/org/zaproxy/zap/extension/api/API.java

```diff
@@ -289,13 +290,17 @@ public boolean handleApiRequest (HttpRequestHeader requestHeader, HttpInputStrea
 
 					ApiResponse res;
 					switch (reqType) {
-					case action:	
-						// TODO Handle POST requests - need to read these in and then parse params from POST body
-						/*
-						if (getOptionsParamApi().isPostActions()) {
-							throw new ApiException(ApiException.Type.DISABLED);
+					case action:
+						if (requestHeader.getMethod().equalsIgnoreCase(HttpRequestHeader.POST)) {
+							if(requestHeader.getHeader(HttpHeader.CONTENT_TYPE)
```

Comment:
This will lead to a NullPointerException if the request does not contain the Content-Type request header. Might worth inverting the equals check.